src/main.rs

```diff
@@ -29,42 +29,22 @@ async fn main() {
         Err(_err) => return,
     };
 
-    let provision_with_password = imds::get_provision_with_password(&imds_body);
-    let disable_authentication = match provision_with_password {
+    if match imds::get_provision_with_password(&imds_body) {
         Ok(disable_authentication) => disable_authentication,
         Err(_err) => return,
-    };
-
-    let username;
-    let mut password = "".to_owned();
-
-    if !disable_authentication {
-        media::make_temp_directory().unwrap();
-
-        media::mount_media();
-
-        let ovf_body = media::read_ovf_env_to_string().unwrap();
-        let environment = media::parse_ovf_env(ovf_body.as_str()).unwrap();
-
-        username = environment
-            .provisioning_section
-            .linux_prov_conf_set
-            .username;
-        password = environment
-            .provisioning_section
-            .linux_prov_conf_set
-            .password;
+    } {
+        // fail if password authentication is enabled.
+        return;
```

Comment:
Ideally we should report the error back to the platform so that provisioning can fail. Otherwise, the user will wait for 20 minutes and get a provisioning timeout, without a clear indication that provisioning failed because of password usage. See [#58](https://github.com/Azure/azure-init/issues/58)

Given that the change to report failure can be scoped to itself, I am ok with merging this one before tackling the other one in a separate PR.

Reply:
Thanks, will try to address the error reporting issue later in a separate PR.